p2p/discovery/mdns/mdns_test.go

```diff
@@ -97,13 +79,13 @@ func TestOtherDiscovery(t *testing.T) {
 	assert.Eventuallyf(
 		t,
 		func() bool {
-			for _, notif := range notifs {
-				infos := notif.GetPeers()
+			for i := 0; i < n; i++ {
+				infos := notifs[i].GetPeers()
 				ids := make([]peer.ID, 0, len(infos))
 				for _, info := range infos {
 					ids = append(ids, info.ID)
 				}
-				if !containsAllHostIDs(ids) {
+				if !containsAllHostIDs(ids,i) {
```

Comment:
Why not just pass the peer ID into this function?

Reply:
Thank you for your suggestion. 
I've revised it. 
Do you think it's ok? 
Please contact me if there is anything else that needs to be modified.